spec/acceptance/mod_passenger_spec.rb

```diff
@@ -113,12 +110,28 @@ class { 'apache::mod::passenger':
       end
     end
     context 'default passenger config' do
-      pp = <<-MANIFEST
-          /* stock apache and mod_passenger */
-          class { 'apache': }
-          class { 'apache::mod::passenger': }
-          #{pp_rackapp}
-      MANIFEST
+      # We need to set passenger_instance_registry_dir on every sane distro
+      # with systemd. Systemd can force processes into a seperate/private
+      # tmpdir. This is the default for apache on Ubuntu 18.04. As a result,
+      # passenger CLI tools can't find the config/socket, which defaults to /tmp
+      # we enable it for ubuntu 16.04/18.04, centos7 and debian 9
+      pp = if ['7', '9', '16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
```

Comment:
These magic numbers feel like a difficult to maintain/easy to break setup. Perhaps it should check for `fact('service_provider') == 'systemd'` instead? You could even do it in a pure puppet block:
```puppet
$passenger_instance_registry_dir = $fact['service_provider'] == 'systemd' ?
  true    => '/var/run',
  default => undef,
}
class { 'apache': }
class { 'apache::mod::passenger':
  passenger_instance_registry_dir => $passenger_instance_registry_dir,
}
```